File: 0094_Binary Tree Inorder Traversal/solution.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
 func inorderTraversal(root *TreeNode) []int {
    if root == nil {
        return nil
    }
    curr := root
    var inorder []int
    for curr != nil {
        if curr.Left == nil {
            inorder = append(inorder, curr.Val)
            curr = curr.Right
        } else {
            prev := curr.Left
            for prev.Right != nil && prev != curr {
                prev = prev.Right
            }
            if prev.Right == nil {
                prev.Right = curr
                curr = curr.Left
            } else {
                prev.Right = nil
                inorder = append(inorder, curr.Val)
                curr = curr.Right
            }
        }
    }
    return inorder
}

File: 0094_Binary Tree Inorder Traversal/solution.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def inorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        inorder = []
        curr = root
        while curr != None:
            if curr.left == None:
                inorder.append(curr.val)
                curr = curr.right
            else:
                prev = curr.left
                while prev.right != None and prev.right != curr:
                    prev = prev.right
                if prev.right == None:
                    prev.right = curr
                    curr = curr.left
                else:
                    prev.right = None
                    inorder.append(curr.val)
                    curr = curr.right
        return inorder